### PR TITLE
chore: add option to login through the new api w/ pre-hashed pwd [DET-5270]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- add a param to /api/v1/auth/login to allow passing presalted and hashed password
- if the username is not provided respond with an invalid argument instead of invalid credentials.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: proto/src/determined/api/v1/auth.proto [hunk 1]
```diff
@@ -15,6 +15,8 @@ message LoginRequest {
   string username = 1;
   // The password of the user.
   string password = 2;
+  // Indicate whether the password is pre-salted & hashed or not.
```
Comment (stoksc) [inline proto/src/determined/api/v1/auth.proto:18]:
```suggestion
  // Indicate whether the provided password is pre-salted & hashed or not.
```
A little less ambiguous.

**Review by stoksc — APPROVED** [review]

**Comment by hamidzr** [comment]
that was fast thanks!

File: proto/src/determined/api/v1/auth.proto (hunk 1 above)
Comment (hamidzr) [reply proto/src/determined/api/v1/auth.proto]: I shoulda committed through the ui 🤦🏼

Comment (stoksc) [reply proto/src/determined/api/v1/auth.proto]: @hamidzr changed this to make sure people know we don't just store the prehashed password ha.

**Pull request merged**